### PR TITLE
Create SaaS doc

### PR DESCRIPTION
**Pull request by ilyapuchka**

### CODE REVIEW

File: Cookbook/Technical-Documents/SaaS.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+- create an asset catalog in `Brand/SaaS/Assets/SaaSConfigFileName` with the required assets (can be empty if no custom assets required)
+- create new AppCenter apps and put their identifiers in the config file
+- create new Firebase apps and download their plists into the `Brand/SaaS/Firebase/SaaSProductName` folder (note that the folder name should match the value of the `product_name` in the config file, which can be different than the config file name itself)
+- generate the app with `bundle exe generate_saas saas: SaaSConfigFileName` and run it
+- commit all the new files but **do not commit changes to generated files**.
+
+## Extending configurations
+
+We should strive to minimise amount of compile time configuration and allow runtime configuration instead so whenever possible consider using product config for the feature configuration. If it's not possible and feature is configured in the AppConfiguration or as a static configuration then to being able to configure it for different SaaS projects you need to:
```
Comment (sdaniel55) [inline Cookbook/Technical-Documents/SaaS.md:85]:
```suggestion
We should strive to minimise amount of compile time configuration and allow runtime configuration instead. Whenever possible consider using product config for the feature configuration. If it's not possible and feature is configured in the `AppConfiguration` or as a static configuration, then to be able to configure it for different SaaS projects, you need to:
```

File: Cookbook/Technical-Documents/SaaS.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+- add the key with the configuration value in the config file. Note that you should add it to all the config files or check if the key is defined in the template
+- if the configuraiton is not yet part of the existing templates extract it into a new template or add to the existing one depending on where it fits better
```
Comment (sdaniel55) [inline Cookbook/Technical-Documents/SaaS.md:88]:
```suggestion
- if the configuration is not yet part of the existing templates, extract it into a new template or add to the existing one depending on where it fits better
```

File: Cookbook/Technical-Documents/SaaS.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+- if the configuration is a part of the static feature configuration then add it as a parameter to the corresponding feature module configuration call in `Brand/SaaS/Templates/SaaSAppDelegate.swift.erb`
```
Comment (sdaniel55) [inline Cookbook/Technical-Documents/SaaS.md:89]:
```suggestion
- if the configuration is a part of the static feature configuration, then add it as a parameter to the corresponding feature module configuration call in `Brand/SaaS/Templates/SaaSAppDelegate.swift.erb`
```

File: Cookbook/Technical-Documents/SaaS.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+- the template files are ERB templates so make sure you use corret [ERB syntax](https://puppet.com/docs/puppet/latest/lang_template_erb.html#concept-5566). So to write the configuration value in the generated file use `<%= cofig_key %>` syntax in the template.
```
Comment (sdaniel55) [inline Cookbook/Technical-Documents/SaaS.md:90]:
```suggestion
- the template files are ERB templates so make sure you use corret [ERB syntax](https://puppet.com/docs/puppet/latest/lang_template_erb.html#concept-5566). To write the configuration value in the generated file, use `<%= cofig_key %>` syntax in the template.
```

File: Cookbook/Technical-Documents/SaaS.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+- if you added a new template file add it to the list of template files in the `generate_saas_files` method in the `fastlane/Lanes/saas`
```
Comment (sdaniel55) [inline Cookbook/Technical-Documents/SaaS.md:91]:
```suggestion
- if you added a new template file, add it to the list of template files in the `generate_saas_files` method in `fastlane/Lanes/saas`
```

File: Cookbook/Technical-Documents/SaaS.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+```
+bundle exec fastlane match_development saas:SaaSConfigFileName
+bundle exec fastlane match_enterprise saas:SaaSConfigFileName
+```
+
+- create an asset catalog in `Brand/SaaS/Assets/SaaSConfigFileName` with the required assets (can be empty if no custom assets required)
+- create new AppCenter apps and put their identifiers in the config file
+- create new Firebase apps and download their plists into the `Brand/SaaS/Firebase/SaaSProductName` folder (note that the folder name should match the value of the `product_name` in the config file, which can be different than the config file name itself)
+- generate the app with `bundle exe generate_saas saas: SaaSConfigFileName` and run it
+- commit all the new files but **do not commit changes to generated files**.
```
[the remark this answers is not in the document]
Comment (ilyapuchka) [reply Cookbook/Technical-Documents/SaaS.md:81]: Not unless there is a need for target specific localisations, but I have left it out as it's more than just that

**Review by danspndl — APPROVED** [review]
👌

**Review by zopa-simon-cass — APPROVED** [review]
🎆

**Pull request merged**